### PR TITLE
Refactor Card and FactCheckCard

### PR DESCRIPTION
**Pull request by dariusk**
I'm considering `Card` now to be an empty rectangle container that can be passed a `cardUrl` and a `footer`, and highlights when you mouse over it. The card manages an `isHovered` state that is provided to its `children` via `Context` as needed. So in this case `FactCheckCard` renders a `Card` with some children in the correct layout. The `ItemDescription` is wrapped with the `CardHoverContext.Consumer`, because `ItemDescription` needs to know that `Card` (not `FactCheckCard`!) is being hovered. In other cases I would do this by passing state from Card to its children, but since we have a generic `children` object I'm doing it this way instead. Seems to work well.

References: CV2-3991

## Type of change

- [X] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

### CODE REVIEW

**Review by caiosba — APPROVED** [review]
Code looks good! I like that now `<Card />` is way simpler. I tested it locally with the fact-checks cards and it works well (I had to push a fix though, which I talked to @dariusk before). The only issue I noticed is that ellipsis are now missing for long URLs:

Captura de tela de 2023-12-22 17-54-47

But I'm approving for now in order to unblock, but please fix it in the next PR, @dariusk.

**Comment by dariusk** [comment]
@caiosba I made a bunch of updates -- please see CV2-3991 comments for details but basically I have added:

 * sorted list of platforms that requests came from for a `SharedItemCard` (called `ItemChannels`)
 * helper function for compact number formatting (and longer number formatting)
 * conditional media icon when the media count is 1
 * tooltips for workspaces, dates, and counts - these are pretty complex and I explain more in CV2-3991

File: src/app/components/cds/media-cards/ItemChannels.js
Comment (caiosba) [inline src/app/components/cds/media-cards/ItemChannels.js:55]: Question/suggestion: What if we move this mapping to https://github.com/meedan/check-web/blob/develop/src/app/CheckChannels.js? We do something similar in https://github.com/meedan/check-web/blob/develop/src/app/CheckFeedLicenses.js.

It can be out of scope for this ticket too - maybe we can just create a small refactoring ticket to abstract that to https://github.com/meedan/check-web/blob/develop/src/app/CheckChannels.js and replace it in other places it's done (like the channels filter).

File: src/app/components/cds/media-cards/ItemChannels.js
Comment (caiosba) [inline src/app/components/cds/media-cards/ItemChannels.js:98]: I like `exact` instead of `shapeOf`!

File: src/app/components/search/SearchResultsCards/SharedItemCard.js
Comment (caiosba) [inline src/app/components/search/SearchResultsCards/SharedItemCard.js:190]: Missing `propTypes`:

- `mediaThumbnail`
- `mediaType`
- `workspaces`

**Review by caiosba — CHANGES_REQUESTED** [review]
@dariusk , looks good! I added a couple of comments, and locally I noticed that a truncation bug (I think it's related to the URL) still happens (the URL should truncate with ellipsis):

pr-1785-bug-1

File: src/app/components/cds/media-cards/ItemChannels.js
Comment (dariusk) [reply src/app/components/cds/media-cards/ItemChannels.js:55]: Good idea. Filed CV2-4181 for the refactor.

**Comment by dariusk** [comment]
> @dariusk , looks good! I added a couple of comments, and locally I noticed that a truncation bug (I think it's related to the URL) still happens (the URL should truncate with ellipsis):
> 
> pr-1785-bug-1

@caiosba It looks like this is on some part of the Check app -- this component is not anywhere except for `Sandbox` right now, and it appears to be truncating correctly there:

image

You'll notice in the above screenshot that when the width is low, 1) the ellipsis truncation happens, and 2) the elements stack in a responsive flex flow.

**Comment by caiosba** [comment]
Thanks @dariusk , can you please try with a long URL? I see it working with long titles and summaries in the fact-check card too.

**Comment by dariusk** [comment]
@caiosba Long url does break it, looking into it.

**Comment by dariusk** [comment]
I fixed the bug with displaying fact check urls that are very long, and I merged @Caio Almeida's work for `CheckFeedDataPoints`, adding the logic for the different display states for `SharedFeedItemCard` based on the data composition of a shared feed:

image

I added three switches to the sandbox - two of them to toggle what kind of data points are in a shared feed, and one to toggle the number of fact checks between 1 and 12345.

File: src/app/components/Sandbox.js
Comment (caiosba) [inline src/app/components/Sandbox.js:533]: Just a reminder to delete this before merging

File: src/app/components/Sandbox.js
Comment (caiosba) [inline src/app/components/Sandbox.js:537]: Just a reminder to delete this before merging

**Review by caiosba — APPROVED** [review]
Thanks @dariusk , looks good now!

**Pull request merged**
